core/templates/components/graph-services/graph-layout.service.spec.ts

```diff
@@ -0,0 +1,1207 @@
+      },
+      {
+        source: {
+          depth: 0,
+          offset: 0,
+          reachable: true,
+          x0: 0.07250000000000001,
+          y0: 0.15333333333333335,
+          xLabel: 0.1625,
+          yLabel: 0.23333333333333334,
+          id: 'State1',
+          label: 'State1',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        target: {
+          depth: 1,
+          offset: 1,
+          reachable: true,
+          x0: 0.29750000000000004,
+          y0: 0.42000000000000004,
+          xLabel: 0.3875,
+          yLabel: 0.5,
+          id: 'State3',
+          label: 'State3',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        d: 'M0.23 0.31333333333333335 Q 0.30557165934031566' +
+          ' 0.3408718290982754 0.32 0.42'
+      },
+      {
+        source: {
+          depth: 1,
+          offset: 0,
+          reachable: true,
+          x0: 0.07250000000000001,
+          y0: 0.42000000000000004,
+          xLabel: 0.1625,
+          yLabel: 0.5,
+          id: 'State2',
+          label: 'State2',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        target: {
+          depth: 2,
+          offset: 0,
+          reachable: true,
+          x0: 0.07250000000000001,
+          y0: 0.6866666666666666,
+          xLabel: 0.1625,
+          yLabel: 0.7666666666666666,
+          id: 'State4',
+          label: 'State4',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        d: 'M0.1625 0.5800000000000001 Q 0.2025 0.6333333333333333' +
+          ' 0.1625 0.6866666666666665'
+      },
+      {
+        source: {
+          depth: 1,
+          offset: 1,
+          reachable: true,
+          x0: 0.29750000000000004,
+          y0: 0.42000000000000004,
+          xLabel: 0.3875,
+          yLabel: 0.5,
+          id: 'State3',
+          label: 'State3',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        target: {
+          depth: 2,
+          offset: 0,
+          reachable: true,
+          x0: 0.07250000000000001,
+          y0: 0.6866666666666666,
+          xLabel: 0.1625,
+          yLabel: 0.7666666666666666,
+          id: 'State4',
+          label: 'State4',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        d: 'M0.32 0.5800000000000001 Q 0.30557165934031566' +
+          ' 0.6591281709017246 0.23000000000000004 0.6866666666666665'
+      }
+    ];
+
+    let returnedAugmentedLinks = sgls.getAugmentedLinks(nodeData, links2);
+    for (var i = 0; i < returnedAugmentedLinks.length; i++) {
+      expect(returnedAugmentedLinks[i].d).toBe(expectedAugmentedLinks[i].d);
```

Comment:
There's too much data here, which obscures what you're trying to check. Can you just validate the correctness of the Bezier curves as stated in the test description?

Reply:
The test description only states that a returned augmented link must have a Bezier curve. To check if the Bezier curves are correct or not, we would need to implement the logic in the test (which may not be ideal). Is there any other way to check if a Bezier is correct?

Reply:
Yup, just make a list of the expected 'd' values and compare the 'd' values to their expected ones (you don't need to catalog all the other fields in the expected values). Alternatively/additionally, you could verify that each returned augmented link has a d value and that it is formatted as 'M%f %f Q ...' (please fill in the rest of the regex).

Reply:
Made changes to check only the list of 'd' values. I think if we are already checking if the 'd' values match the expected values, there would be no point in checking if those values match the format (as they are already equal to the correctly formatted values). What do you think?

Reply:
Updates the changes, will add a comment in the next push